### DISCUSSION TITLE
failed to build on OS X Ventura 13.3 with latest Xcode and Developer Tools

### DISCUSSION THREAD

**Discussion by fakahle** [iraf-community/iraf#272] [General]
IRAF failed to build on OS X 13.3 (latest Xcode and Developer Tools installed):
```
cc -c -O -w  -g -Wall -O2  -I/opt/iraf/iraf/include -DNOVOS sysdep.c
sysdep.c:112:3: error: call to undeclared function 'rmdir'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                rmdir(s);
                ^
sysdep.c:112:3: note: did you mean 'rmtdir'?
sysdep.c:107:1: note: 'rmtdir' declared here
rmtdir(Void)
^
1 error generated.
make[1]: *** [sysdep.o] Error 1
make: *** [sysgen] Error 2
```
[build.1.log](https://github.com/iraf-community/iraf/files/11133051/build.1.log)

Adding `#include "unistd.h"` (to unix/f2c/src/sysdep.c) seems to fix this issue, but unfortunately a linker issue pops up later on:
```
Undefined symbols for architecture x86_64:
  "_aclrc_", referenced from:
      _xcallc_ in libsys.a(calloc.o)
      _xmfree_ in libsys.a(mfree.o)
      _mallo1_ in libsys.a(malloc1.o)
      _ffault_ in libsys.a(ffault.o)
  "_aclri_", referenced from:
      _vfnopn_ in libsys.a(vfnmap.o)
  "_aclrs_", referenced from:
      _envint_ in libsys.a(envinit.o)
  "_amovc_", referenced from:
      _vfndel_ in libsys.a(vfnmap.o)
      _vfnenr_ in libsys.a(vfnmap.o)
      _krealc_ in libsys.a(krealloc.o)
      _kiopes_ in libsys.a(kiopenks.o)
      _ktzget_ in libsys.a(ktzget.o)
      _ktzput_ in libsys.a(ktzput.o)
      _kiflux_ in libsys.a(ktzput.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
[build.2.log](https://github.com/iraf-community/iraf/files/11133057/build.2.log)

Since I've a running IRAF installation on my M1-MacBook (build and compiled in 2022) I retried with those sources ... same error (build.1.log) as above:-( From this I would guess that these issues are somehow related to XCode and Developer Tools (tried to build with 14.3 and 14.2). Anybody else experiencing problems with a fresh installation on Ventura 13.3?

Do I miss something from the logs?

**Comment by olebole** [comment]
I can confirm this and also went into this problem today after I updated XCode. It seems that the updated clang 
is a bit more picky now. Unfortunately, I don't have constant access to a Mac, so it may take a while to fix it.

I converted this to bug report #273

**Reply by fakahle** [reply]
At first sight it looked like patching iraf.h (adding `#include <string.h>` and `#include <stdlib.h>`) might fix the linker issues, but there are plenty more (like fe this one:
```
1 error generated.
errs.c:326:7: error: call to undeclared function 'zdojmp_'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                longjmp (errenv, 1);
                ^
/opt/iraf/iraf/include/iraf/setjmp.h:17:37: note: expanded from macro 'longjmp'
#define longjmp(e,v)    (u_jmpstat=(v),ZDOJMP((e),&u_jmpstat))
```
